fastestimator/trace/io/test_report.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import inspect
+import json
+import os
+from typing import Callable, List, Union
+
+from fastestimator.trace.trace import Trace
+from fastestimator.util.data import Data
+from fastestimator.util.traceability_util import traceable
+from fastestimator.util.util import to_list
+from time import time
+from datetime import datetime
+import pdb
+
+@traceable()
+class TestReport(Trace):
+    """Automate testing and report generation.
+
+    Args:
+        test_descriptions: List of text-based descriptions.
+        test_criterias: List of test functions. Function input argument names needs to match keys from the data
+            dictionary.
+        test_title: Title of the test.
+        json_output: Path into which to write the output results JSON.
+        doc_output: Path into which to write the output QMS summary report (docx).
```

Comment:
unused doc parameter. and these two params should probably be consolidated into 1

Reply:
fixed